crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -56,6 +59,24 @@ pub async fn router(state: StateType) -> Router {
                                     }
                                 }))
                             }
+                            MethodInvocation::Hardhat(HardhatMethodInvocation::SetBalance(address, balance)) => {
```

Comment:
Instead of this becoming an enormous `match` statement with all logic, could you split it using function calls?

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/3998/commits/7ab3fd0879636121fb43112fc6a0c19f3b91981c